### PR TITLE
initial role

### PR DESCRIPTION
**Pull request by e0d**
Role to deploy the edx-notes-api application

@maxrothman @fredsmith @feanil

### CODE REVIEW

File: playbooks/roles/edxlocal/tasks/main.yml [hunk 1]
```diff
@@ -110,6 +110,20 @@
     priv='{{ ANALYTICS_API_CONFIG['DATABASES']['reports']['NAME'] }}.*:SELECT'
   when: ANALYTICS_API_CONFIG is defined
 
+- name: setup the edx-notes-api db user
```
Comment (carsongee) [inline playbooks/roles/edxlocal/tasks/main.yml:113]: You mind adding these [here](https://github.com/edx/configuration/blob/master/playbooks/roles/mariadb/defaults/main.yml) too?

File: playbooks/roles/edx-notes-api/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+#
+# This data structure will be written out to yaml configuration file
+# in /edx/etc
+#
+SERVICE_CONFIG:
+  ALLOWED_HOSTS: "{{ EDX_NOTES_API_ALLOWED_HOSTS }}"
+  # replace with your secret key
+  SECRET_KEY: '{{ EDX_NOTES_API_SECRET_KEY }}'
+  # replace with your oauth id and secret
+  CLIENT_ID: "edx-notes-id"
```
Comment (tymofij) [inline playbooks/roles/edx-notes-api/defaults/main.yml]: CLIENT_ID and CLIENT_SECRET should also be configurable

File: playbooks/roles/edx-notes-api/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  # Max number of rows to return in result.
+  RESULTS_MAX_SIZE: 250
+  DATABASE_OPTIONS:
+    connect_timeout: 10
+  DATABASES: "{{ EDX_NOTES_API_DATABASES }}"
+  HAYSTACK_CONNECTIONS:
+    default:
+      ENGINE: 'notesserver.highlight.ElasticsearchSearchEngine'
+      URL: "{{ EDX_NOTES_API_ELASTICSEARCH_URL }}"
+      INDEX_NAME: 'edx-notes'
```
Comment (maxrothman) [inline playbooks/roles/edx-notes-api/defaults/main.yml]: Why can't this just reference the same thing as `ELASTICSEARCH_INDEX`?

File: playbooks/roles/edx-notes-api/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      INDEX_NAME: '{{ EDX_NOTES_API_DATASTORE_NAME }}'
+  DISABLE_TOKEN_CHECK: True  
+#
+# vars are namespace with the module name.
+#
+edx_notes_api_role_name: "edx-notes-api"
+edx_notes_api_service_name: "edx-notes-api"
+# underscore name
+edx_notes_api_role_name_safe: "edx_notes_api"
+
```
Comment (maxrothman) [inline playbooks/roles/edx-notes-api/defaults/main.yml]: Are all three of these being defined separately necessary?

**Comment by tymofij** [comment]
cc @explorerleslie

**Comment by e0d** [comment]
@carsongee Can you verify that I've done what you need?  @maxrothman @fredsmith thumbs?

File: playbooks/roles/mariadb/defaults/main.yml [hunk 5]
```diff
@@ -78,6 +78,11 @@ MARIADB_USERS:
     priv: "*.*:CREATE USER"
     host: "{{ MARIADB_HOST_PRIV }}"
 
+  - name: "{{ EDX_NOTES_API_MYSQL_DB_USER|default('notes001') }}"
+    pass: "{{ EDX_NOTES_API_MYSQL_DB_PASS|default('secret') }}"
+    priv: "*.*:ALL"
```
Comment (carsongee) [inline playbooks/roles/mariadb/defaults/main.yml]: the priv here gives `notes001` super user privs instead of just privs on their own DB

**Comment by maxrothman** [comment]
Superseded by a forthcoming PR from max/merge-refactor

**Pull request closed without merging**